src/firestore/indexes.ts

```diff
@@ -59,16 +59,17 @@ export class FirestoreIndexes {
       );
     }
 
-    indexesToDeploy.forEach(async (index) => {
+    const indexPromises: Array<Promise<any>> = [];
+    for (const index of indexesToDeploy) {
```

Comment:
TIL: If you do `forEach` with an `async` function inside it you just generate a big fat race condition.

Reply:
`forEach` is sync, so you could `await` your `async` function in `forEach` (but this is just silly):

```typescript
indexesToDeploy.forEach(await (async (index) => { /* do stuff */ }));
```

The alternative to this `forEach`-populating-an-array is to use `map` and return the promise at the end, then `Promise.all` them (though, it looks like you have conditionals here, so maybe `forEach` is correct!)